src/base/static/components/molecules/map-layer-selector.js

```diff
@@ -0,0 +1,83 @@
+import React from "react";
+import PropTypes from "prop-types";
+import classNames from "classnames";
+import Spinner from "react-spinner";
+
+import { Icon, InfoModalTrigger } from "../atoms/feedback";
+import "./map-layer-selector.scss";
+
+const MapLayerSelector = props => {
+  let status;
+  if (!props.layerStatus || !props.layerStatus.isVisible) {
+    status = "hidden";
+  } else {
+    status = props.layerStatus.status;
+  }
+
+  return (
+    <div className="map-layer-selector">
+      <span
+        className="map-layer-selector__selectable-area"
+        onClick={() => props.onToggleLayer(props.layerId)}
+      >
+        <Icon icon={props.icon} classes="map-layer-selector__id-icon" />
+        <span
+          className={classNames("map-layer-selector__layer-title", {
+            "map-layer-selector__layer-title--selected": props.selected,
+          })}
+        >
+          {props.title}
+        </span>
+        {status === "loading" && props.layerStatus.isVisible ? (
+          <div className="map-layer-selector__spinner-container">
+            <Spinner />
+          </div>
+        ) : (
+          <Icon
+            icon={classNames({
+              "fa-check": status === "loaded",
+              "fa-times": status === "error",
+            })}
+            classes={classNames("map-layer-selector__status-icon", {
+              "map-layer-selector__status-icon--green": status === "loaded",
+              "map-layer-selector__status-icon--red": status === "error",
+            })}
+          />
+        )}
+      </span>
+      <InfoModalTrigger
```

Comment:
Does this need to be a modal? I think hover text here would be sufficient and a more appropriate UX.

Reply:
Yeah, the modal was kind of the low hanging fruit implementation. I agree a flyout or hover text would probably be better here.